### DISCUSSION TITLE
Changing GitHub user after moving repositories to new user

### DISCUSSION THREAD

**Discussion by Shavve** [codemagic-ci-cd/codemagic-docs#2265] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

We have recently moved our repository to a new GitHub user, which mean that the repository URL is also changed. I cannot find how to change it in Codemagic.

### Expected results

To be able to change the repository URL on my app so it will fetch the repo from the new user and all of our workflow will work the same without creating the workflow again from scratch.

### Actual results

Seem like there is no option to do the switch.

### Build id (optional)

6579d15e24d0a653beeb25cb

**Comment by dtrdic** [comment, marked as answer]
Hi there, i'll change the repo path from my end, but in order to do so;
​﻿
﻿1. Please share with me the AppId of the application whose path needs to be changed
﻿- you can find the appID inside the URL
﻿
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/b90d222d-910a-4e64-9077-2040559c2bc2)




﻿2 Add a new application to Codemagic - the one that is now in a different location and share the app ID with me
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/351c5a7c-1b40-4f25-936a-b84db91652ec)



﻿Then I will make the appropriate changes on my end.

**Reply by Shavve** [reply]
Hi David @dtrdic ,

Thank you for the quick reply.

1. The AppID - 64424b810fcb982cd2ec1711
2. Why we need to create a new App ? we have only one app (the appID I provided). We want only to change the repository to the new github account. If we will create a new app we will need to configure all the integration, certificates, secrets and workflows from scratch - and we want to avoid that.

**Reply by dtrdic** [reply]
hey @Shavve you don't have to configure anything on this second app, I will just need the AppID from the second app as well in order to make a migration.

**Reply by Shavve** [reply]
Ok, got it.
But I cannot select my repo from the drop-down when trying to create new app.
<img width="766" alt="Screenshot 2023-12-14 at 15 14 33" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/149502588/eecb018b-d2cf-4e5e-b4f2-4cb37fdce888">

The integration to GitHub is working fine, but CodeMagic new-app-wizard not showing the repos.

Also, it seems that the current existing app (AppID: 64424b810fcb982cd2ec1711) is missing information that we already configured in the past and it worked. In addition, we cannot see any build history, except one failed build from yesterday.

**Reply by dtrdic** [reply]
maybe your repo has some additional restrictions, let's add it again but this time let's use the "Other" option and clone your repo with SSH key​
1. Click on Add application
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/61753549-2e7e-4042-8813-bcbcd49982e7)

2. Select team and connect repo using the "Other" option
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/8de73dbf-79f1-4e0b-889b-d5f07dc72e1a)

3. Clone your repo using SSH option
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/a81d6b69-9cbc-4f97-bbec-6d7002361f68)

4. Click on generate key button and copy
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/ed5836da-e08a-49c4-809d-4f5b03da4d0a)

5. Go to https://github.com/settings/keys and add new SSH key

**Reply by iamkoami** [reply]
Hello I have the issue I want to change the repo url
The AppID - 64f87a03cbc8012dc5a69fba
I also create a new app link to the new repo 
The AppID - 65802a1d537791ffa92466bf

Please what is the next step for migration ?
Best regards.

**Reply by dtrdic** [reply]
hey @iamkoami, migrated, can you try to run a test build?

**Reply by iamkoami** [reply]
@dtrdic that work perfectly, Thanks for the support.
